PoC/runtimes/kafka/kafka-consumer/src/main/java/org/eclipse/tractusx/edc/kafka/consumer/EdrTokenCallbackHandler.java

```diff
@@ -0,0 +1,177 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.edc.kafka.consumer;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.security.auth.AuthenticateCallbackHandler;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerToken;
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerTokenCallback;
+import org.apache.kafka.common.security.oauthbearer.internals.secured.BasicOAuthBearerToken;
+import org.apache.kafka.common.security.oauthbearer.internals.secured.ConfigurationUtils;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.AppConfigurationEntry;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.kafka.common.config.SaslConfigs.SASL_OAUTHBEARER_SCOPE_CLAIM_NAME;
+import static org.eclipse.tractusx.edc.kafka.consumer.KafkaConsumerApp.ASSET_ID;
+
+/**
+ * Callback handler that retrieves an OAuth bearer token from EDRData and converts it to
+ * a format suitable for Kafka SASL/OAUTHBEARER authentication.
+ * <p>
+ * This handler processes OAuthBearerTokenCallback instances by:
+ * 1. Retrieving the access token from an EDR provider
+ * 2. Parsing the JWT token to extract required claims
+ * 3. Creating a BasicOAuthBearerToken that Kafka can use for authentication
+ */
+@Slf4j
+public class EdrTokenCallbackHandler implements AuthenticateCallbackHandler {
+
+
+    private boolean isInitialized = false;
+    private String scopeClaimName;
+    private final EdrDataProvider edrDataProvider;
+
+    /**
+     * Constructor with default EdrDataProvider implementation
+     */
+    public EdrTokenCallbackHandler() {
+        this.edrDataProvider = new DefaultEdrDataProvider();
+    }
+
+    /**
+     * Constructor with custom EdrDataProvider for testing
+     *
+     * @param edrDataProvider custom provider implementation
+     */
+    public EdrTokenCallbackHandler(final EdrDataProvider edrDataProvider) {
+        this.edrDataProvider = edrDataProvider;
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs, final String mechanism, final List<AppConfigurationEntry> jaasConfigEntries) {
+        ConfigurationUtils cu = new ConfigurationUtils(configs, mechanism);
+        scopeClaimName = cu.get(SASL_OAUTHBEARER_SCOPE_CLAIM_NAME);
+        isInitialized = true;
+    }
+
+    @Override
+    public void handle(final Callback[] callbacks) throws UnsupportedCallbackException, IOException {
+        checkInitialized();
+
+        for (Callback callback : callbacks) {
+            if (callback instanceof OAuthBearerTokenCallback tokenCallback) {
```

Comment:
add logging in case instanceof does not match.

Reply:
An exception is thrown if callback does not match